crypto/cmp/cmp_vfy.c

```diff
@@ -326,19 +326,23 @@ int OSSL_CMP_print_cert_verify_cb(int ok, X509_STORE_CTX *ctx)
     return (ok);
 }
 
-/* return 1 if expiration should be checked and the given time has passed */
-int OSSL_CMP_expired(const ASN1_TIME *endtime, const X509_VERIFY_PARAM *vpm)
+/* return 0 if time should not be checked or reference time is within frame,
+   or else 1 if it s past the end, or -1 if it is before the start */
+int OSSL_CMP_cmp_timeframe(const ASN1_TIME *start,
+                           const ASN1_TIME *end, const X509_VERIFY_PARAM *vpm)
 {
     time_t check_time, *ptime = NULL;
-    unsigned long flags = X509_VERIFY_PARAM_get_flags((X509_VERIFY_PARAM*)vpm);
+    unsigned long flags = vpm == NULL ? 0 :
+                          X509_VERIFY_PARAM_get_flags((X509_VERIFY_PARAM*)vpm);
 
     if ((flags & X509_V_FLAG_USE_CHECK_TIME) != 0) {
         check_time = X509_VERIFY_PARAM_get_time(vpm);
         ptime = &check_time;
+    } else if ((flags & X509_V_FLAG_NO_CHECK_TIME) != 0) {
+        return 0; /* ok */
     }
-
-    return (!(flags & X509_V_FLAG_NO_CHECK_TIME) &&
-            endtime && X509_cmp_time(endtime, ptime) < 0);
+    return (end   != NULL && X509_cmp_time(end  , ptime) < 0) ? +1 :
```

Comment:
nested "?" operators might be confusing, consider 

if (end   != NULL && X509_cmp_time(end  , ptime) < 0) {
 return +1;
} else if (start != NULL && X509_cmp_time(start, ptime) > 0) {
return -1;
} else {
 return 0;
}

Reply:
Good point, done